### PR TITLE
Refactor/index changes

### PR DESCRIPTION
**Pull request by F-Camuso**
### Overview
Moving permissions and schemas out of the index file

### Pull request type
- [ ] Bug fixes;
- [ ] New feature;
- [X] Refactoring;
- [ ] Documentation;
- [ ] Other: __________________

### Current behaviour
There's a lot of functions inside the index file that should have a separated file

### New behaviour

index will be a lot clear now.

The permissions folder will have all the rules and the shield function.

### Additional informations

Its a work in progress...

### CODE REVIEW

**Review by SantiagoMateus — APPROVED** [review]

**Comment by EduardoGiacomini** [comment]
I am thinking about this refactor. Maybe we can broken this `src/index.ts` in some folders and files:

```
- src
    - config (all setups)
        environment.ts (exposes used environment)
        logger.ts (**we need do add a logger to this app**)
    - resolvers (our custom resolvers)
    - generated (auto generated resolvers)
    - application or server (all files to create the server, middlewares, and more)
        - middlewares
        - resolvers (expose resolvers used by application)
        application.ts or server.ts (the class who will create the server app)
        index.ts (only exposes the application/server)
    index.ts (only create a new application/server instance and run the app)
```

What do you think about?

**Comment by F-Camuso** [comment]
> I am thinking about this refactor. Maybe we can broken this `src/index.ts` in some folders and files:
> 
> ```
> - src
>     - config (all setups)
>         environment.ts (exposes used environment)
>         logger.ts (**we need do add a logger to this app**)
>     - resolvers (our custom resolvers)
>     - generated (auto generated resolvers)
>     - application or server (all files to create the server, middlewares, and more)
>         - middlewares
>         - resolvers (expose resolvers used by application)
>         application.ts or server.ts (the class who will create the server app)
>         index.ts (only exposes the application/server)
>     index.ts (only create a new application/server instance and run the app)
> ```
> 
> What do you think about?

looks good to me, i will work on this after doing the routes

File: src/config/logger.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import winston from 'winston';
+// import { ENVIRONMENT } from './environment';
```
Comment (EduardoGiacomini) [inline src/config/logger.ts:2]: Opa

File: src/permissions/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    Query: {
+      users: rules.isAdmin,
+      patients: rules.isAuthenticated,
+      examGroups: rules.isAuthenticated,
+    },
+  },
+  {
+    fallbackError: async (thrownThing, parent, args, context, info) => {
+      Logger.info(thrownThing);
```
Comment (EduardoGiacomini) [inline src/permissions/index.ts:29]: Logger.error

File: src/resolvers/patient-x-exams/CreatePatientXExamsResolver.ts [hunk 3]
```diff
@@ -65,7 +66,7 @@ export default class CreatePatientXExamsResolver {
         });
       }
     }
-    // console.log(patientXExams);
+    Logger.info(patientXExams);
```
Comment (EduardoGiacomini) [inline src/resolvers/patient-x-exams/CreatePatientXExamsResolver.ts:69]: debug

File: src/routes/reports/exams.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            },
+          },
+        },
+      },
+    },
+  };
+
+  if (examCategoryId) {
+    query.where = query.where || {};
+    Logger.info(examCategoryId);
```
Comment (EduardoGiacomini) [inline src/routes/reports/exams.ts:32]: debug

File: src/routes/reports/index.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import express from 'express';
+import examsCsv from './exams';
+import patientsCsv from './patients';
+import patientsExamsCsv from './patientsExams';
+
+const reports = express.Router();
+
+reports.use('/exams', examsCsv);
+reports.use('/patients', patientsCsv);
+reports.use('/patientsExams', patientsExamsCsv);
```
Comment (EduardoGiacomini) [inline src/routes/reports/index.ts:10]: lindo

File: src/routes/reports/patientsExams.ts [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+      exame: a._name,
+      'custo unitário': utils.formatter(value.currentValue),
+      'categoria do exame': a._category_name,
+      'grupo do exame': a._category_examGroup_name,
+      ocorrências: value.occurrences,
+      'valor total': utils.formatter(value.occurrences * value.currentValue),
+    };
+    return obj;
+  });
+  Logger.info(singlePatientExams);
```
Comment (EduardoGiacomini) [inline src/routes/reports/patientsExams.ts:122]: debug

File: src/schema.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  PatientCrudResolver,
+  ExamsCrudResolver,
+  ExamCategoryCrudResolver,
+  ExamGroupCrudResolver,
+  PatientXExamsCrudResolver,
+  IndividualUseGroupCrudResolver,
+  IndividualUseGroupXMeasurementUnitCrudResolver,
+  MeasurementUnitCrudResolver,
+  IndividualUseCrudResolver,
+  PatientXIndividualUseCrudResolver,
```
Comment (EduardoGiacomini) [inline src/schema.ts:14]: opa

**Review by EduardoGiacomini — CHANGES_REQUESTED** [review]
Very good

**Review by brunomond — APPROVED** [review]

**Pull request merged**